Issue #13155: update the docs example for block initialization methods

Z4ID-Reaper opened the pull request:
references #13155

Z4ID-Reaper commented:
@Zopsss @romani  please review, apologies for the delay.

also please see #16798 , once it has the `approve `label i can start working on it.

romani commented on src/site/xdoc/checks/coding/variabledeclarationusagedistance.xml:
```diff
@@ -98,42 +98,54 @@ public class Example1 {
-          Check can detect a block of initialization methods. If a variable is used in
-          such a block and there are no other statements after variable declaration,
-          then distance = 1.
+          Check is smart to detect a block of initialization methods as a single point for distance.
```
Check can detect ...

romani commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/Example6.java:
```diff
@@ -0,0 +1,36 @@
+  }
+
+  public void calSetAgain(long timeNow, int hh, int min){
+    int minutes = min + 5000;  // violation, distance = 6
```
`// violation 'Distance between variable ''minutes'' declaration and its first usage is 6`

romani replied:
Bad examples will be updated in scope of https://github.com/checkstyle/checkstyle/issues/16881

Z4ID-Reaper replied:
can i use multiline ? 
both of these are giving 85 character line length limit violation.

```
// violation 'Distance between variable ''minutes'' declaration and its first usage is 6
// violation below, 'Distance between variable ''minutes'' declaration and its first usage is 6
```

romani replied:
`// violation below, 'variable ''minutes'' declaration and its first usage is 6`

romani reviewed (changes requested):
Items

Pankraz76 commented on src/site/xdoc/checks/coding/variabledeclarationusagedistance.xml:
```diff
@@ -98,42 +98,54 @@ public class Example1 {
-          Check can detect a block of initialization methods. If a variable is used in
-          such a block and there are no other statements after variable declaration,
-          then distance = 1.
+          Check is smart to detect a block of initialization methods as a single point for distance.
```
```suggestion
          Check is able to detect a block of initialization methods as a single point for distance.
```
`is able to` or `is capable of` might be a more common approach.

Z4ID-Reaper replied:
done

Pankraz76 commented on src/site/xdoc/checks/coding/variabledeclarationusagedistance.xml:
```diff
@@ -98,42 +98,55 @@ public class Example1 {
-          Check can detect a block of initialization methods. If a variable is used in
-          such a block and there are no other statements after variable declaration,
-          then distance = 1.
+          Check can detect a block of initialization methods as a single point for distance.
```
```suggestion
          Check can detect a block of initialization methods, as a single point for distance.
```

Z4ID-Reaper replied:
done

Pankraz76 reviewed (changes requested):
spelling `,`

Z4ID-Reaper commented:
Ill fix this and and other issues too , once i get to my laptop tomorrow. 
I am out of station at this moment :)

Z4ID-Reaper commented:
@romani @Pankraz76 updated , please review for merge.

Pankraz76 commented on src/site/xdoc/checks/coding/variabledeclarationusagedistance.xml:
```diff
@@ -98,42 +98,55 @@ public class Example1 {
-        <p>
-          Case #1:
+        <p id="Example6-config">
+          configure the check to default:
```
```suggestion
          Configure the check to default:
```
If this is a new sentence, then should start with UpperCase. In `L:103` we completed the previous one with using `Dot.`.
Other `<p>` elements start like this as well.

Z4ID-Reaper replied:
done

Pankraz76 reviewed (changes requested):
last bit, then merge. +1

Pankraz76 reviewed (approved):
thanks

Zopsss commented:
Github, generate site

Z4ID-Reaper commented:
Github, generate site

romani commented on src/site/xdoc/checks/coding/variabledeclarationusagedistance.xml.template:
```diff
@@ -41,42 +41,32 @@
           <param name="type" value="code"/>
         </macro>
         <p>
-          Check can detect a block of initialization methods. If a variable is used in
```
pelase add `<hr class="example-separator"/>` before `<p>`, to make it clear where new example is starting.

Z4ID-Reaper replied:
Done

romani reviewed (changes requested):
last

Zopsss commented on src/site/xdoc/checks/coding/variabledeclarationusagedistance.xml.template:
```diff
@@ -39,44 +39,34 @@
         </p>
-        <p>
-          Case #1:
+        <p id="Example6-config">
```
Example 6 is after Example 1

Please rename example 6 to be example 2, rename all other preceding examples and increase their number by one.

Z4ID-Reaper replied:
done

Zopsss commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/Example2.java:
```diff
@@ -1,38 +1,38 @@
-    System.out.println("Statement 3");
-    num = 1;
-    PI = 3.14;
+  public void calSet(long timeNow, int hh, int min) {
```
```suggestion
  public void case1(long timeNow, int hh, int min) {
```

Z4ID-Reaper replied:
done

Zopsss commented on src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/Example2.java:
```diff
@@ -1,38 +1,38 @@
-      b = 2;
-      count++;
-    }
+  public void calSetAgain(long timeNow, int hh, int min){
```
```suggestion
  public void case2(long timeNow, int hh, int min){
```

Z4ID-Reaper replied:
done

Zopsss reviewed (changes requested):
Last:

Z4ID-Reaper commented:
@Zopsss 
![Screenshot 2025-04-25 090428](https://github.com/user-attachments/assets/88e21e32-7ded-4a34-a08e-b07d27cd221e)

should i change this to 

`Specify maximum distance between declaration of variable and its first usage. Values should be greater than 0.`

Z4ID-Reaper commented:
Github, generate site

Pankraz76 commented:
Yes please, nice idea.

plural:
`Specify maximum distance between declaration of variable and its first usage. Values should be greater than 0.`
singular:
`Specify maximum distance between declaration of variable and its first usage. Value should be greater than 0.`
`Specify maximum distance between declaration of variable and its first usage. Value should be positive, greater than 0.`
`Specify maximum distance between declaration of variable and its first usage. Value should be positive (greater than 0).`

there is only one param and not a collection, right? So we might want to apply singularity as well.

Pankraz76 commented:
May improve spelling by using , using some articles:
`Specify the maximum distance between the declaration of a variable and its first usage. Value should be greater than 0.`
And or make things more compact: `declaration and its first usage`
`Specify the maximum distance between a variable´s declaration and its first usage. Value should be greater than 0.`

Z4ID-Reaper commented:
Github, generate site

Pankraz76 reviewed (approved):
awesome, thank you.

Z4ID-Reaper commented:
Github, generate site

Z4ID-Reaper commented:
### Summary of changes made so far

- updated the example for block initialization as shown in example 2
- updated the violation message comments
- updated the description of `allowedDistance `, `validateBetweenScopes ` properties

this pr is in a position to close the following issues :- #13155 , #16881 

from my side it's all good 👍.

Zopsss commented:
This PR was initially targeting only 1 issue right? Now it's targeting 2 issues. 

@romani should we proceed with these changes? Or should we split the PR?

Z4ID-Reaper commented:
The second issue was very small work, and very related, to the point it arised from this PR only. 
So i included both.

Z4ID-Reaper commented:
@romani your final review is awaited, PR is complete from my side.

romani commented:
@Z4ID-Reaper , sorry for being late, please resolve conflicts.

Z4ID-Reaper commented:
Github, generate site

romani reviewed (approved):
thanks a lot.
